src/java_bytecode/java_types.h

```diff
@@ -374,4 +374,45 @@ void get_dependencies_from_generic_parameters(
   const typet &,
   std::set<irep_idt> &);
 
+class java_specialised_generic_class_typet : public java_generic_class_typet
+{
+public:
+  java_specialised_generic_class_typet()
+  {
+    set(ID_C_specialised_generic_java_class, true);
+  }
+
+  ~java_specialised_generic_class_typet()
+  {
+  }
+};
+
+inline bool java_is_specialised_generic_class_type(const typet &type)
+{
```

Comment:
IIRC the correspondant predicates are named `is_java_generic_FOO_BAR`, please check and change name here to adapt